### ISSUE TITLE
[DateTimePicker] Endless Loop / Freezing Browser on invalid moment input

### ISSUE THREAD

**Issue by BGehrels** [mui/mui-x#5821]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Current behavior 😯

When i click on the calender icon in a DateTimePicker, the screen freezes and my cpu fan starts blowing.

### Expected behavior 🤔

The calendar picker should open

### Steps to reproduce 🕹

- Use a DateTimePicker with date-io/moment:2.15.0, moment 2.29.4 and  moment.js and @mui/x-date-pickers:5.0.0-beta.5
- set the value to `moment('123')` or `moment(null)` which creates an invalid moment object
- click the calender icon in the Picker
- The screen freezes while the picker loops endlessly in the folowing loop in `date-util.js` (`findClosestEnabledDate()`): `while (forward || backward) {}`

### Your environment 🌎

  System:                                                  
    OS: Linux 5.10 Debian GNU/Linux 10 (buster) 10 (buster)
    CPU: (3) x64 Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz  
    Memory: 2.92 GB / 4.82 GB                              
    Container: Yes                                         
    Shell: 5.0.3 - /bin/bash                               
  Binaries:                                                
    Node: 16.16.0 - /usr/local/bin/node                    
    Yarn: 1.22.19 - /usr/local/bin/yarn                    
    npm: 8.11.0 - /usr/local/bin/npm                       
  Managers:                                                
    Apt: 1.8.2.3 - /usr/bin/apt                            
  Utilities:                                               
    Make: 4.2.1 - /usr/bin/make
    GCC: 8.3.0 - /usr/bin/gcc
    Git: 2.20.1 - /usr/bin/git
    Mercurial: 4.8.2 - /usr/bin/hg
    Subversion: 1.10.4 - /usr/bin/svn
  Languages:
    Bash: 5.0.3 - /bin/bash
    Perl: 5.28.1 - /usr/bin/perl
    Python: 2.7.16 - /usr/bin/python
    Python3: 3.7.3 - /usr/bin/python3

**Comment by alexfauquette** [comment]
I tried with both
```jsx
const [value, setValue] = React.useState(moment(null));
// and
const [value, setValue] = React.useState(moment('123'));
```
And I do not manage to reproduce your issue. Did you use some date validation prop such as `minDate`?

https://codesandbox.io/s/basicdatetimepicker-demo-mui-x-forked-ws5few?file=/demo.tsx

**Comment by BGehrels** [comment]
Here You'll find a list of all probs supplied:
![image](https://user-images.githubusercontent.com/1008320/185144442-6811ee45-a8c6-4c40-b48b-ef189e398e12.png)

**Comment by BGehrels** [comment]
Removing the `date` Prob (that seems not to be part of the official signature) fixes the problem. Interestig....

**Comment by BGehrels** [comment]
Here is the codesandbox to reproduce it:
https://codesandbox.io/s/basicdatetimepicker-demo-mui-x-forked-trpn6w?file=/demo.tsx

- Wait for it to render
- Set Focus to the Input field
- Clear it and type "1"
- Klick the calendar icon
- Freeeeeeze

**Comment by alexfauquette** [comment]
`date` and `rawValue` are not part of the signature and `date` is effectively the source of your bug.

When rendering the calendar, we provide to him the `date={parsedValue}` with `parsedValue` either a valid date or `null`. we also extends remaining props with `{...other}`. Since you provide an invalid `date` prop, this one overrides the parsed value

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/internals/components/CalendarOrClockPicker/CalendarOrClockPicker.tsx#L175:L184

I'm not sure we can do something for that

**Comment by BGehrels** [comment]
Yep, they seemed to be leftovers of past refactorings/signature changes. 

> I'm not sure we can do something for that

From a users perspective, i would expect that parameters that are not in the spec
- trigger failures,
- are silently ignored or 
- trigger warnings and are ignored afterwards.

I'm not sure if `{...other}` is a good pattern in this case (or in general when it comes to component signatures) - the signature of the component is it's contract with the public, so why not pass the properties that are part of the signature explicitly to child components?

Side note:
ESLint has [a rule for that](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/jsx-props-no-spreading.md) for good reasons:

> It is also good for maintainability by avoiding passing unintentional extra props and allowing react to emit warnings when invalid HTML props are passed to HTML elements.

wich is exactly the situation we see here.

**Comment by LukasTy** [comment]
@alexfauquette What do you think about putting on L176 in your example, essentially not allowing user to override the `date` field at least? 🤔

**Comment by alexfauquette** [comment]
About using  `...other` seems the main reason is to save package size.

https://github.com/mui/mui-x/blob/master/packages/x-date-pickers/src/StaticDatePicker/StaticDatePicker.tsx#L72:L74
```js
// Note that we are passing down all the value without spread.
// It saves us >1kb gzip and make any prop available automatically on any level down.
```

A second reason is customizability. We can not anticipate all the customizations developers will do. So we pass props that are necessary for the default behavior, and let `{...other}` at the end to allow developers to pass new props or override default ones.

I thaugh Proptypes would catch this kind of problem. But it seems to be able to find wrong prop types but not prop which are not defined. But typescript detects that the prop is not part of the component signature.

@LukasTy If it's to prevent this issue, we can also check that the `date` is valid in `findClosestEnabledDate` and throws an error if not.

**Comment by alexfauquette** [comment]
By the way, #6128 will prevent your bug by removing `date`

**Comment by flaviendelangle** [comment]
Fixed by #5821
